backend/apps/events/signals.py

```diff
@@ -0,0 +1,15 @@
+from apps.events.models import Event
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+@receiver(
+    post_save,
+    sender=Event,
+    dispatch_uid="calculate_duration_for_event",
+)
+def calculate_duration_for_event(sender, instance, **kwargs):
+    """This calculates duration of the event after each change."""
+    new_duration = instance.end_time - instance.start_time
+    object = Event.objects.filter(pk=instance.id)
+    object.update(duration=new_duration)
```

Comment:
You don't need to make a query to a database here, because you already have an Event object as the parameter `instance`, so you can just do:
```
instance.update(duration=instance.end_time-instance.start_time)

```

Reply:
Or just remove the django signal, then override the model's method `save()` and set the value there before saving the instance.